### ISSUE TITLE
Discussion of documentation of Vue components

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#427]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This issue is to discuss and potentially implement the first instances of documentation `<docs>` blocks within Vue single file components (SFCs). Information on custom blocks can be found [here](https://vuejs.org/api/sfc-spec.html#language-blocks), and info on docs blocks can be found [here](https://vue-loader.vuejs.org/guide/custom-blocks.html#example).

Would be great to get the community's feedback on this and if we'd like to put quick bits on usage in the `<docs>` blocks :)

### Contribution

Would be happy to discuss this with people! Implementing any needed settings changes and adding in the first few `<docs>` blocks could also be a great `good first issue` 😊

**Comment by andrewtavis** [comment]
@ahmedy00, it'd be interesting to get your feedback on this as you've been working with Vue longer than most of us 😊 Have you used `<docs>` tags before?

**Comment by ahmedy00** [comment]
@andrewtavis I haven't used the <docs> tag in any project before, so I don't have educated guess as to the benefits it can provide.
In my opinion it may seem beneficial for new contributors at first glance but it may cause code complexity, readability and compatibility problems in the long run.

**Comment by andrewtavis** [comment]
Yes, I was thinking about also the need to maintain all of this 🤔 The whole point for Vue is that it has the structure of HTML as a basis and should be easy to understand given good naming criteria. We'll also need to do documentation eventually as well, which is where we'd send people 🤔 My one question is, would there be a way to auto document components? Have you heard of/used anything for that? Would be cool if we could generate something from the code and just make sure that we write proper comments where needed.

**Comment by andrewtavis** [comment]
Thanks for the feedback!

**Comment by andrewtavis** [comment]
Not sure if you'd have interest in #424, btw @ahmedy00. I know that you prefer `script-template-style`, but to me/the team as we're using Tailwind it really does make sense to have template first as we don't have style blocks. If you wanted to set up the linting for that it'd be a lot of help 😊

**Comment by ahmedy00** [comment]
@andrewtavis The reason I use `script-template` order is that the VueJS community has been using it more often recently. I understand that this is not a better fit for the activist than the `template-script`, for the reasons you stated in #424. I would enjoy working for #424

**Comment by andrewtavis** [comment]
Sounds good, @ahmedy00! Appreciate the feedback that what we're doing makes sense 😊 Feel free to write in #424 and I'll assign you :)

**Comment by andrewtavis** [comment]
Hey @bmartins-42 👋 I was going to close this issue as @ahmedy00 and I decided against `<docs>` blocks that this issue was originally about, but then I was thinking that this might be a good place for the discussion of Storybook. @ahmedy00 would be someone I'd want in on the discussion as well. Would you want to use this one to pitch the ideas you had on that?

**Comment by bmartins-42** [comment]
My idea was to implement a tool like [storybook](https://storybook.js.org/) that would let us access components in UI interface where we can play with props and event. It would simplify the discoverability of components available for dev and non-dev contributors.

There is some plugin/configuration already available for [Nuxt](https://storybook.nuxtjs.org/), [tailwind](https://storybook.js.org/recipes/tailwindcss) or [i18n](https://storybook.js.org/addons/storybook-i18n).

The downside is that we will need to write/update [stories](https://storybook.js.org/docs/react/get-started/whats-a-story) to describe components in storybook. Also, tooling will need some setup configuration/maintenance with package updates.

I have never worked on implementing storybook, so I do not have any personal feedback. If anyone have experience with and could share it, that would be great.

<hr />

Other reading:

https://scalingo.com/blog/guide-storybook-vue-js

**Comment by andrewtavis** [comment]
Is definitely interesting, @bmartins-42 :) Have you used it before for work or another project?

**Comment by andrewtavis** [comment]
Ability to link the Figma designs and have them as a tab in the story is soooo clean. Thinking about this, if we're working on a new component we'd potentially be able to make a story for it and link the Figma component and then someone literally has the design sitting there for comparison as they're working on the component and using Storybook for reference. Also making sure that we have designs for all components. 

Will check alternatives now to make sure we're doing our homework, but this seems so cool.

**Comment by andrewtavis** [comment]
There's also a [Nuxt module](https://github.com/nuxt-modules/storybook) for it :) Only alternative I saw was [Histoire](https://github.com/histoire-dev/histoire), which should get a look in that it's from a Vue Core Team member. Would allow us to customize it more, but then the integrations aren't there yet.

**Comment by andrewtavis** [comment]
But also looking at Histoire, really like how easy it is to theme it and the fact that we're just writing Vue Single File Components for the stories 👍👍 Obviously Storybook is more developed, but if we get Histoire in the DevTools and a Figma plugin eventually this could be much easier for us to maintain.

**Comment by bmartins-42** [comment]
> Is definitely interesting, @bmartins-42 :) Have you used it before for work or another project? 

Unfortunately no, only read about it.

> Ability to link the Figma designs and have them as a tab in the story is soooo clean

I didn't see this part, but indeed this would help a lot. [Here are some doc about](https://storybook.js.org/blog/figma-plugin-for-storybook/)

If you validate storybook as a documentation, i'm interested to do the setup.

**Comment by andrewtavis** [comment]
How do you feel about Histoire though, @bmartins-42? Maybe waiting a bit on this until they get auto documentation up could be something? I'm not expecting it'll be too long off 🤔 I could contact the maintainers and see if they could provide an estimate on that, if that's the main thing we're missing. 

I'm just a bit worried about unnecessarily adding a different syntax into the codebase when there is something that does job and uses our current technologies. We could very easily make good first issues for people to document the components where they're just writing a Vue SFC, then afterwards they're also ready to actually start contributing code to the project. 

I don't think that this is something we can decide on before the next dev sync on the 4th, so I'll plan on bringing it up then! You'd be more than welcome to attend! Details will be on Matrix 😊

**Comment by andrewtavis** [comment]
And thank you so much for the offer to set up Storybook! Really appreciate the willingness to take on such a large task 🚀 Let's figure this out and get some great documentation out that can help new people come in and build great things!

**Comment by bmartins-42** [comment]
So I did take a look at Histoire, it's promising, but I don't think it's ready yet. 
I also regret the fact they didn't leverage SFC with <doc> or <story> tag in template, but it's probably due to the way SFC is transformed.

I understand your concern about creating syntax for storybook, especially if we want the documentation to be a gate for newcomers. 
I guess we need to discuss furthermore or make a POC to compare the result.

I will try to attend the next dev sync :)

**Comment by andrewtavis** [comment]
Great that you'll try to be at the next sync, @bmartins-42! Already have this point typed into the topics for discussion 😊 Let's be in touch on this, and if it means we need to wait for a bit to see if Histoire develops a bit then we can wait. Aside from this we can of course use Storybook if the need becomes more immediate.

**Comment by andrewtavis** [comment]
Checked out [Histoire](https://github.com/histoire-dev/histoire) again given that [Nuxt Devtools just hit v1.0](https://nuxt.com/blog/nuxt-devtools-v1-0) a bit ago. There's still no module for Histoire, but it does appear to be on the roadmap for the organization as in the 1.0 release docs they're mentioning integrations between the dev tools and Histoire 😊

**Comment by jb90000** [comment]
Hey @andrewtavis, I notice this issue is tagged as good for first-timers so I'm chiming in here! I see lots going on in the comments here - is there any work left outstanding on this issue that requires documentation support?

**Comment by andrewtavis** [comment]
Hey @jb90000 👋 This would be something good to work on at some point, but we're a bit blocked with other issues at the moment. What's needed here is to test the setup we have for Histoire and then to start making the first frontend component stories. I'd be happy to explain more if that'd be something you'd be interested in working on :)

**Comment by jb90000** [comment]
Thanks, @andrewtavis ! Would those tasks be suitable for a non-technical first-timer? If so, I'd love to learn more. 

For context, I'm a writer making my foray into GitHub as a way of practicing docs as code flows.

**Comment by andrewtavis** [comment]
Yes, they certainly would be! Give me a moment to get back from vacation and a few weeks to get the MVP out, but then we can start with this? I can also give it a look next weekend and see if the baseline can get up and running :)

**Comment by jb90000** [comment]
Sounds great - enjoy the rest of your vacation!

**Comment by jb90000** [comment]
Hey @andrewtavis, figured I'd follow up here to see if this is still on your radar. I know you mentioned next steps would be to test the Histoire setup and then to start making the first frontend component stories. 

Let me know if there's anything I can do to help get things started!

**Comment by andrewtavis** [comment]
Appreciate you checking in here, @jb90000! Let me get through the current PRs and we should be good to start looking into this :) Hopefully next week?

**Comment by jb90000** [comment]
Sounds great, @andrewtavis - I'll keep an eye out for your response!

**Comment by Suz99** [comment]
Hi @andrewtavis!  after reading the documentation links provided, to configure the vue-loader to recognize and process these `<docs>` blocks, we might need to specify additional webpack loaders if the documentation requires special processing like markdown or other formats. 

please let me know if this suggestion is useful.

**Comment by andrewtavis** [comment]
Big thing for this is that I need to turn on and set up Histoire for this, but I've been really overloaded and didn't have the time... Sorry about this, @jb90000 :( We need to get a few more things done and then update the frontend dependencies, but then we should be able to start in with this. Interesting to see that Histoire still isn't stable as well though 🤔

**Comment by Imthiaz123** [comment]
Hi @andrewtavis ,
I've come across this under the good first issue tag and I'd like to know the status here. As a newbie in terms of open source, I'd love to take part in this if you provided the opportunity

**Comment by andrewtavis** [comment]
Thanks for checking in on this, @Imthiaz123 :) We're still waiting for Histoire v1 as a full release before we work on this. Let's hold up for now and then we can do the setup and make a bunch of issues to document individual components 😊

**Comment by shubhamzombade72** [comment]
Hi @andrewtavis 👋  
  I’ve been following this discussion and understand that this is currently waiting on Histoire v1.  
  Once it’s ready, I’d be happy to help with the setup or start working on documenting the first few Vue components.  
  Please let me know if I can take this issue or if there’s any prep work I can do meanwhile.  
  Thanks!

**Comment by andrewtavis** [comment]
Thanks so much, @shubhamzombade72! Sorry for getting back to you a bit late. Has been quite busy at the end/start of the year.

Looks like Histoire hit v1.0 beta just a few days ago, so we should be good to go here soon 😊 See the releases: https://github.com/histoire-dev/histoire/releases. I'll assign the issue to you. Would you want to do some research on what the first steps would be given that we can be pretty sure that things will be consistent now?

Note that we do have the `histoire.config.ts` and `histoire.setup.ts` files, with the contents as of now being commented out. Would be great to know if these files need to change :)

Others in the thread would also be welcome to support! Please let us know if you'd like to contribute.

**Comment by jangjohn** [comment]
Hi @andrewtavis and team! 

I'd love to contribute to this issue. I'm familiar with Vue 3 and TypeScript from my work on React + TypeScript projects, and I'm interested in helping set up the `<docs>` blocks for the components.

A few questions to make sure I understand correctly:
1. Are there specific components you'd like to prioritize for the first documentation blocks?
2. Should the `<docs>` blocks include usage examples or just high-level descriptions?
3. Is there an existing documentation style guide I should follow?

Looking forward to your guidance and happy to start with a small PR once we align on the approach! 😊

**Comment by andrewtavis** [comment]
Hey @jangjohn 👋 Thanks for checking in here :) Would you want to help @shubhamzombade72 work on this? I think we've been waiting on this for enough time, and we're at v1.0 beta now, so hopefully there wouldn't be too much that changes? Do you two want to open a PR that uncomments the code in the `histoire*` files in [frontend](https://github.com/activist-org/activist/tree/main/frontend) and sets up a basic working component registry for the project?

From there it'd be great if we document how to get it all set up!

Please let us know what you'd like to work on :) Same goes to others on this issue!

**Comment by andrewtavis** [comment]
From there we can make a new issue where we start documenting components 📝

**Comment by Technoritesh152005** [comment]
I'd like to work on this project

**Comment by andrewtavis** [comment]
We're still in holding for histoire v1.0.0 that's not in beta. Once this is done we can begin on :) Thank you for your interest!

**Comment by cheralakeerthana07** [comment]
hey! i'm intrested in contributing with the issue. I'd like to start by identifying few Vue SFC's where short documentation usage is used, then investigate the required cofiguration for <docs> custom blocks. Would it be oky if i worked on the initial implementation with few example components?

**Comment by andrewtavis** [comment]
Thanks for your interest, @cheralakeerthana07 :) I realized that this issue should be blocked. We're quite focused on the first release and don't really have capacity to look into other things. If you find another issue, then please let us know!
